src/lib/Client.ts

```diff
@@ -0,0 +1,57 @@
+import fetch from "node-fetch";
+import ResponseError from "./ResponseError";
+import { APIResponse } from "./types";
+
+class Client {
+  private _apiKey: string;
+
+  constructor(apiKey: string) {
+    if (!apiKey) {
+      throw new Error("No API key provided");
+    }
+
+    this._apiKey = apiKey;
+  }
+
+  public async lookup(ip: string): Promise<APIResponse> {
+    try {
+      const response = await fetch(
+        `https://whois.as207111.net/api/lookup?ip_address=${encodeURI(ip)}`,
+        {
+          method: "GET",
+          headers: {
+            Accept: "application/json",
+            Authorization: `Bearer ${this._apiKey}`,
+          },
+        }
+      );
+
+      if (response.status !== 200) {
+        switch (response.status) {
```

Comment:
Maybe we should also pass some details to the ResponseError, like the response.message, errors like 409, 503 have a message defined.

Reply:
the messages were taken from the data you sent on the discord server, if the statusCode is not in the switch case, it does send the response data. a more nice way would be to just pass the errors field. I'll look into it